src/components/ChatEntry.js

```diff
@@ -1,22 +1,40 @@
-import React from 'react';
-import './ChatEntry.css';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp'; 
+import './ChatEntry.css';
+
+const ChatEntry = ({ sender, body, timeStamp, liked, onLikeChange, isLocal}) => {
+  const [isLiked, setLiked] = useState(liked); 
+
+  const handleLikeToggle = () => {
+    setLiked((prevLiked) => !prevLiked); 
+    onLikeChange(!isLiked);
+  };
 
-const ChatEntry = (props) => {
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={`chat-entry ` + (isLocal ? 'local': 'remote')}>
+      <h2 className="entry-name">{sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{body}</p>
+        <p className="entry-time">
+          <TimeStamp time={timeStamp} />
+        </p>
+        <button 
+            className="like"
+            onClick={handleLikeToggle}>
+            {isLiked ? '❤️' : '🤍'} 
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool.isRequired,
+  onLikeChange: PropTypes.func.isRequired,
```

Comment:
Because this proptype is marked `isRequired`, we're getting warnings in the tests, since the test author didn't know what you were going to call this property, and so it's absent from the props passed in during the tests. To avoid the warnings, we can leave off `isRequired`.

Ideally, we'd also include code to ensure that any "optional" values are actually present before trying to use them. For example, in the like click handler we could do something like
```js
    if (onLikeChange) {
        onLikeChange(id);  // (id is currently not being passed to this component, but should be)
    }
```
which would only try to call the function if it's not falsy (and `undefined` is falsy).

Reply:
👀 We should pass the `id` of the message into the ChatEntry(both in the destructuring and the prop types here). Ideally, this component shouldn't need to know how to do anything besides telling some other code that a message's liked button was clicked. An easy way to achieve that is by including the `id` in the props so that when the liked button is clicked, we can pass the `id` to the callback we receive to handle that.